src/propmachinery.jl

```diff
@@ -1,32 +1,45 @@
-import Base: start, next, done
-
 abstract QuPropagatorMethod
 
-immutable QuPropagator{QPM<:QuPropagatorMethod}
-    hamiltonian
-    init_state::QuVector
+immutable QuPropagator{QPM<:QuPropagatorMethod, QVM<:Union(QuBase.AbstractQuVector,QuBase.AbstractQuMatrix)}
+    eq
```

Comment:
I think I don't understand your doubts? We only store subtypes of `QuEquation`, which might be created in the constructor.